### PR TITLE
Reorganized fixes for Wish

### PR DESCRIPTION
**Pull request by BardurArantsson**
I decide to just merge 'next', and I've taken the liberty of rebasing + changing things around a bit + setting you as author. I've only done very minimal changes in the commits, and I've reworded the commit messages a bit to fit.

Could you please give your approval @tkaitchuck for merging these? I've added a comment to one of the commits where I'm a bit unsure if it's right (your original code).

 Could you also, just for-the-record state that you'd be OK with relicensing your contribution under the GPL? Very unlikely to happen, but just in case... :)

### CODE REVIEW

File: src/cmd5.cc [hunk 1]
```diff
@@ -1879,7 +1879,8 @@ bool is_ok_spell(s32b spell_idx, s32b pval)
 	}
 	// Are we permitted to cast based on item pval? Only music
 	// spells have non-zero minimum PVAL.
-	if (pval < spell_type_minimum_pval(spell))
+	s32b min_pval = spell_type_minimum_pval(spell);
+	if (min_pval > 0 && pval < min_pval)
```
Comment (BardurArantsson) [inline src/cmd5.cc]: This is where I'm a bit unusure... Should this be min_pval >= 0, or is > correct?

**Comment by BardurArantsson** [comment]
Btw, thanks for the fixes. The wish thing has been bugging me since forever, but it just never got high enough on my list of priorities since I've mostly played Anti-magic users :)

File: src/cmd5.cc (hunk 1 above)
Comment (tkaitchuck) [reply src/cmd5.cc]: Per the comment, only music spells have pvals != 0 and we only want to disable music spells so > is correct.

**Comment by BardurArantsson** [comment]
Great, thank you!

**Comment by BardurArantsson** [comment]
Could also state FTR that you'd be ok with relicensing as GPL? (If that ever happens, ... see above)

**Comment by tkaitchuck** [comment]
yes

**Pull request merged**